processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/OutOfOrderCommitControl.java

```diff
@@ -90,43 +73,17 @@ public synchronized DeferredCompletion reportFetchedOffset(long offset) {
                     String.format("offsets count overflow: cap=%d, offset=%d", capacity, offset));
         }
 
-        OffsetState state = new OffsetState(offset, false);
+        OffsetState state = new OffsetState(offset);
         states.addLast(state);
-        latest = state.offset;
+        latest = state.offset();
 
-        return () -> complete(state);
+        state.completion().future().whenComplete((unused, throwable) -> onComplete(offset)); // TODO okay in this order?
```

Comment:
So this `whenComplete` call is the 1st one for this CF, and the CF will get several more `whenComplete` calls including ones supplied from user-supplied code. Since the these registered callbacks are called by LIFO order, I was wondering if logging at that last step of CF completion handling could cause misleading timestamp that is delayed by preceding callbacks.